src/Datadog.Trace/ExtensionMethods/StringExtensions.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Globalization;
+
+namespace Datadog.Trace.ExtensionMethods
+{
+    internal static class StringExtensions
+    {
+        /// <summary>
+        /// Removes the trailing occurrence of a substring from the current string.
+        /// </summary>
+        /// <param name="value">The original string.</param>
+        /// <param name="suffix">The string to remove from the end of <paramref name="value"/>.</param>
+        /// <returns>A new string with <paramref name="suffix"/> removed from the end, if found. Otherwise, <paramref name="value"/>.</returns>
+        public static string TrimEnd(this string value, string suffix)
+        {
+            if (value == null) { throw new ArgumentNullException(nameof(value)); }
+
+            return !string.IsNullOrEmpty(suffix) && value.EndsWith(suffix)
+                       ? value.Substring(0, value.Length - suffix.Length)
+                       : value;
+        }
+
+        public static ulong? TryParseUInt64(this string value)
```

Comment:
Well, I can't disagree. I have written similar comments in your PRs 😉 I guess I use the extension  out of habit to avoid the clunky bool/`out` pattern, before inline `out` declarations were a thing.

Removed in a1a3fa210f9a49907892e4766933094623d5e91f.